cron/update_geolite_db.py

```diff
@@ -34,24 +36,29 @@ def fetch_db_data(db_download_location):
         return data
 
 
-def save_db_data(save_path, db_data):
-    temp_save_path = save_path + ".tmp"
-    logger.debug("Saving db data to " + temp_save_path)
-    with open(temp_save_path, "wb+") as text_file:
-        text_file.write(db_data)
-    logger.debug("Testing temp geolite db file")
-    geo_ip = GeoIP(temp_save_path)
-    # Do a generic lookup to verify we don't get any errors (malformed data)
-    geo_ip.lookup_ip('8.8.8.8')
-    logger.debug("Moving temp file to " + save_path)
-    os.rename(temp_save_path, save_path)
+def save_db_data(db_file, db_data):
+    save_path = path.join(options.db_store_location, db_file)
+    fd, temp_path = mkstemp(suffix='.tmp', prefix=db_file, dir=options.db_store_location)
+    with open(temp_path, 'wb') as temp:
+        logger.debug("Saving db data to " + temp_path)
+        temp.write(db_data)
+        fsync(temp.fileno())
+        temp.flush()
+        logger.debug("Testing temp geolite db file")
+        geo_ip = GeoIP(temp_path)
+        # Do a generic lookup to verify we don't get any errors (malformed data)
+        geo_ip.lookup_ip('8.8.8.8')
+        logger.debug("Moving temp file to " + save_path)
+    temp.close()
```

Comment:
`temp` will be closed by the context manager that it exists inside of as a result of being created with `with`.  Calling this method a second time may cause an error.